include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -143,6 +145,16 @@ class SharedDataFacade final : public BaseDataFacade
     {
         BOOST_ASSERT_MSG(!m_coordinate_list.empty(), "coordinates must be loaded before r-tree");
 
+        const auto file_index_ptr = data_layout->GetBlockPtr<char>(
+            shared_memory, storage::SharedDataLayout::FILE_INDEX_PATH);
+        file_index_path = boost::filesystem::path(file_index_ptr);
+        if (!boost::filesystem::exists(file_index_path))
+        {
+            util::SimpleLogger().Write(logDEBUG) << "Leaf file name " << file_index_path.string();
+            throw util::exception("Could not load leaf index file. "
+                                  "Is any data loaded into shared memory?");
```

Comment:
Can we re-word this error message?  Maybe "OSRM shared memory refers to a required file that doesn't exist: filename".